src/librustdoc/html/format.rs

```diff
@@ -281,36 +283,24 @@ crate fn print_where_clause<'a, 'tcx: 'a>(
 
                 match pred {
                     clean::WherePredicate::BoundPredicate { ty, bounds, bound_params } => {
-                        let bounds = bounds;
-                        let for_prefix = if bound_params.is_empty() {
-                            String::new()
-                        } else if f.alternate() {
-                            format!(
-                                "for&lt;{:#}&gt; ",
-                                comma_sep(bound_params.iter().map(|lt| lt.print()), true)
-                            )
-                        } else {
-                            format!(
-                                "for&lt;{}&gt; ",
-                                comma_sep(bound_params.iter().map(|lt| lt.print()), true)
-                            )
-                        };
+                        let ty_cx = ty.print(cx);
+                        let generic_bounds = print_generic_bounds(bounds, cx);
 
-                        if f.alternate() {
+                        if bound_params.is_empty() && f.alternate() {
+                            write!(f, "{ty_cx:#}: {generic_bounds:#}")
+                        } else if bound_params.is_empty() && !f.alternate() {
+                            write!(f, "{ty_cx}: {generic_bounds}")
+                        } else if f.alternate() {
```

Comment:
I'm not a big fan of writing conditions like this. In here, we have one big check: `bound_params.is_empty()` and one "inner" check: `f.alternate()`. So I think like this would make more sense:

```suggestion
                        if bound_params.is_empty() {
                            if f.alternate() {
                                write!(f, "{ty_cx:#}: {generic_bounds:#}")
                            } else {
                                write!(f, "{ty_cx}: {generic_bounds}")
                            }
                        } else {
                            if f.alternate() {
```

Reply:
Committed the requested change.